functions/go/gatekeeper/README.md

```diff
@@ -4,29 +4,93 @@
 
 <!--mdtogo:Short-->
 
-Validate the KRM resources using [Gatekeeper] policies.
+The `gatekeeper` function wraps the [`Gatekeeper`] policy enforcement engine to
+evaluate [Gatekeeper] constraints to validate the KRM resources.
 
 <!--mdtogo-->
 
-### FunctionConfig
+It provides a very flexible and powerful way to validate KRM resources against
+custom policies.
+
+You can learn more about how to use the `Gatekeeper` project [here][howto]. The
+concept document about the `OPA Constraint Framework` can be found
+[here][concept].
 
 <!--mdtogo:Long-->
 
-[Gatekeeper] allows users to validate the KRM resources against the Gatekeeper
-policies.
+### Authoring Policies
+
+The `gatekeeper` function follows the [executable configuration] pattern. There
+are 2 kinds of resources needed to define a policy and they need to be provided
+using `input items` along with other KRM resources to be validated.
+
+- [Constraint Template]: Define the schema and logic of a policy. The policy
+  logic in a Constraint Template must be written in the [Rego] language.
+- [Constraint]: Signal the Gatekeeper the corresponding constraints need to be
+  enforced. Every Constraint must be backed by a Constraint Template.
+
+The constraint templates and the constraints must be provided using
+`input items` along with other KRM resources to be validated. Nothing need to be
```

Comment:
I am wondering if we should just say that: "The constraint templates and the constraints resources should be in the package containing the KRM resources."

I think we can remove the sentence about `input functionConfig`.